core-common/src/main/java/org/zalando/nakadi/repository/kafka/PartitionsCalculator.java

```diff
@@ -1,188 +1,30 @@
 package org.zalando.nakadi.repository.kafka;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.google.common.annotations.VisibleForTesting;
 import org.zalando.nakadi.config.NakadiSettings;
 import org.zalando.nakadi.domain.EventTypeStatistics;
 
-import java.io.IOException;
-import java.io.InputStream;
-import java.util.Arrays;
-import java.util.Map;
-import java.util.NavigableMap;
-import java.util.Objects;
-import java.util.TreeMap;
-import java.util.stream.Stream;
-
 public class PartitionsCalculator {
-    // Contains mapping from message size (in bytes) to list of throughput of kafka with (index + 1) as partition count.
-    private final NavigableMap<Integer, float[]> stats = new TreeMap<>();
     private final int maxPartitionCount;
     private final int defaultPartitionCount;
 
-    private PartitionsCalculator(final InstanceInfo instanceInfo,
-                                 final int defaultPartitionCount,
+    public PartitionsCalculator(final int defaultPartitionCount,
                                  final int maxPartitionCount) {
         this.defaultPartitionCount = defaultPartitionCount;
         this.maxPartitionCount = maxPartitionCount;
-        for (final SpeedStatistics ss : instanceInfo.getStats()) {
-            stats.put(ss.getMessageSize(), ss.getSpeed());
-        }
     }
 
     public int getBestPartitionsCount(final EventTypeStatistics stat) {
-        if (null == stat) {
+        if (null == stat || stat.getReadParallelism() == 0 && stat.getWriteParallelism() == 0) {
```

Comment:
`readParallelism` and `writeParallelism` are already annotated as :

```
    @NotNull
    @Min(value = 1, message = "at least one reader expected")
    private Integer readParallelism;
    @NotNull
    @Min(value = 1, message = "at least one writer expected")
    private Integer writeParallelism;
    ```

Reply:
thats true but I thought since the class is named `PartitionsCalculator.java` and the method is public I would have these additional checks. I updated the check with `<=` in a new change to accomodate for more.

Reply:
That is true, and if we don't run validator explicitly we don't gain protection by using `@Min` annotation.